packages/design-system/src/components/web-components/ds-alert.test.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import Alert, { AlertProps } from '../Alert/Alert';
+import { UtagContainer } from '../analytics';
+import { setAlertSendsAnalytics } from '../flags';
+import { render, screen } from '@testing-library/react';
+import register from 'preact-custom-element';
+
+register(Alert, 'ds-alert');
+
+// Typically, defining this interface would allow for kebab-case props, but isn't on Alert for some reason.
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'ds-alert': AlertProps;
+    }
+  }
+}
+
+const defaultText = 'Ruhroh';
+
+function renderAlert(props: AlertProps = {}) {
+  return render(<ds-alert {...props}>{defaultText}</ds-alert>);
+}
+
+function expectHasClass(className: string) {
+  expect(screen.getByRole('region').className).toContain(className);
+}
+
+describe('Alert', function () {
+  // WC renders <slot> in snaps
+  it('renders alert', () => {
+    renderAlert({ id: 'static-id' });
+    const alert = screen.getByRole('region');
+    expect(alert).toMatchSnapshot();
+  });
+
+  it('renders a heading', () => {
+    const heading = 'Error';
+    renderAlert({ heading });
+    expect(screen.getByText(heading).textContent).toEqual(heading);
+  });
+
+  it('appears as an error', () => {
+    renderAlert({ variation: 'error' });
+    expectHasClass('ds-c-alert--error');
+  });
+
+  it('appears as a lightweight alert', () => {
+    renderAlert({ weight: 'lightweight' });
+    expectHasClass('ds-c-alert--lightweight');
+  });
+
+  it('renders additional className and role prop', () => {
+    const className = 'ds-u-test';
+    const role = 'alert';
+    renderAlert({ 'class-name': className, role });
+
+    // Need to query by class selector instead of role.
+    // Role is both a prop and native HTML attr, so it appears in multiple places within the component and is hard to query for.
+    const alert = document.querySelector('.ds-c-alert');
+    expect(alert.className).toContain(className);
+  });
+
+  it('renders HTML children', () => {
+    renderAlert({ children: <p className="ds-text">{defaultText}</p> });
+    const alert = screen.getByRole('region');
+    expect(alert.textContent).toContain(defaultText);
+  });
+
+  it('hides icon', () => {
+    renderAlert({ 'hide-icon': true });
+    expectHasClass('ds-c-alert--hide-icon');
+  });
+
+  it('sets tabIndex when autoFocus is passed', () => {
+    renderAlert({ autoFocus: true });
+    const alert = screen.getByRole('region');
+    expect(alert.tabIndex).toBe(-1);
+  });
+
+  // experiencing similar issues to analytics hook; is it hooks WC has issues with?
+  it.skip('sets tabIndex when alertRef is passed', () => {
```

Comment:
I think this is a great use for `skip`. It's not that it was working, and now it's not, and now we're ignoring it; it's that we haven't gotten to it yet, so the "skipped" totals in the test output serve as a good indicator of our progress.